src/problems/sudoku_Leuni.clj

```diff
@@ -0,0 +1,192 @@
+(ns problems.sudoku-Leuni)
+
+
+;Our state is only the current board
+(defrecord State [board current-pos])
+
+(def start-state (->State [[0 7 5 0 9 0 0 0 6]
+                           [0 2 3 0 8 0 0 4 0]
+                           [8 0 0 0 0 3 0 0 1]
+                           [5 0 0 7 0 2 0 0 0]
+                           [0 4 0 8 0 6 0 2 0]
+                           [0 0 0 9 0 1 0 0 3]
+                           [9 0 0 4 0 0 0 0 7]
+                           [0 6 0 0 7 0 5 8 0]
+                           [7 0 0 0 1 0 3 9 0]]
+
+                           [8 7]))
+
+
+
+; Function   : all-pos (All Possible)
+;
+; Description: Given a state [board current-pos] this
+;              function will check the current position.
+;              If it is non-zero it returns a child with an
+;              increment to its current-pos and no board changes.
+;              If it is zero it will return children with the zero
+;              replaced by a number 1 through 9 with incremented current-pos.
+;
+; Returns    : 9 children with incremented current-pos if the current-pos is non zero
+;            : 9 children with the numbers 1-9 in current-pos. Also increments current-pos
+(defn all-pos [state]
+  ;Temporarily stores the board and row, column information
+  (let [board (:board state)
+        x-pos (first(:current-pos state))
+        y-pos (second(:current-pos state))]
+
+    ;For numbers 1 through 9, x, create a new state with the number at current-pos changed to the new number x + 1
+    ;If current-pos is non-zero create children that have the same board and incremented current-pos
+    (for [x (range 9)] (if (> (get (get board x-pos) y-pos) 0)
+                         (State. board (if (= x-pos 8) [0 (+ 1 y-pos)] [(+ 1 x-pos) y-pos] ))
+
+                         ;Else change the number at curret-pos to x+1
+                         (State. (assoc board x-pos (assoc (get board x-pos)  y-pos (+ 1 x)))
+
+                                 ;If current-pos is [8 8] don't increment, return the same current-pos with the new board
+                                 (if (and (= x-pos 8) (= y-pos 8)) [8 8]
+
+                                   ;Else if current-pos x-pos is 8 set current-pos to [0 (y-pos + 1)
+                                   (if (= x-pos 8) [0 (+ 1 y-pos)] [(+ 1 x-pos) y-pos])))))))
+
+
+; Function   : check-square
+;
+; Description: Given a state [board current-pos] this
+;              function will determine if there are any duplicate values
+;              in the current-pos ([row, col]) square.
+;
+; Returns    : true  - if no duplicates (all elements are distinct)
+;              false - if duplicates (all elements are not distinct)
+(defn check-square[state pos]
+  (let [board (:board state)
+        cur-row (first pos)
+        cur-col (second pos)]
+
+      ; Determine the appropriate 'square column(legal? start-state)'
+      ; by using which column is the current column
+      (def col-square (if (<= cur-col 2) 0
```

Comment:
Another incorrect use of `def`.

Also, note that the logic for `col-square` and `row-square` is essentially identical, which suggests that you really want to pull that out into a function. When you do that, you can simplify the logic there by just taking the (integer) quotient (there's a function for that) of the col/row and 3.